src/dials/array_family/flex_ext.py

```diff
@@ -1297,14 +1299,44 @@ def map_centroids_to_reciprocal_space(
         # reciprocal lattice viewer -> but if we are looking at the crystal
         # coordinate frame we need to look at the experiments independently
 
+        if hasattr(experiments, "detectors"):
+            detectors = experiments.detectors()
+        else:
+            detectors = list(set(expt.detector for expt in experiments))
+        try:
+            test_subcache = self.detector_panel_sels[detectors[0]]
+        except (KeyError, AttributeError):
+            cache_valid = False
+        else:
+            test_key = list(test_subcache.keys())[0]
+            cache_valid = len(test_subcache[test_key]) == len(self)
+        if not cache_valid:
+            self.detector_panel_sels = {det: {} for det in detectors}
+            # Dirty hack: without the following line, the reflection_table
+            # (a boost.python class) will become unpickleable because the
+            # object's python __dict__ is not empty and we have not declared
+            # that the boost-defined __getstate__ method knows about the stuff
```

Comment:
Could we explicitly wipe the cache on pickle (do we even use pickle?)

Or you talking about general serialisation? Even so, we could explicitly dump the cache before we try and serialise the object, or name it in such a way as it can be ignored at that point?